### PR TITLE
[IMP] cetmix_tower_server backend UX updates

### PR DESCRIPTION
**Pull request by ivs-cetmix**
Forward port of https://github.com/cetmix/cetmix-tower/pull/414

- Tag mixin and tag helper commands
- New helper variables in Python commands

Task: 5045

## Summary by CodeRabbit

* **New Features**
  * Added a reusable tag mixin with add/remove/query tag operations.
  * Tag support applied to commands, plans, servers, file templates, and server templates.
  * New Python helper shortcuts for tower models in command contexts.

* **Improvements**
  * Caching for reference lookups now accounts for user/session context.

* **Tests**
  * Added comprehensive tests validating tag mixin behavior and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by GabbasovDinar — APPROVED** [review]

**Review by Aldeigja — APPROVED** [review]
Functional LGTM

File: cetmix_tower_server/models/cx_tower_command.py [hunk 1]
```diff
@@ -83,11 +84,9 @@ class CxTowerCommand(models.Model):
         "If empty, command can be run on all servers",
     )
     tag_ids = fields.Many2many(
-        comodel_name="cx.tower.tag",
         relation="cx_tower_command_tag_rel",
         column1="command_id",
         column2="tag_id",
-        string="Tags",
     )
```
[the remark this answers is not in the document]
Comment (ivs-cetmix) [reply cetmix_tower_server/models/cx_tower_command.py:90]: It doesn't because this field is defined in the mixin.

**Comment by CetmixGitDrone** [comment]
What a great day to merge this nice PR. Let's do it!
Prepared branch [17.0-ocabot-merge-pr-428-by-ivs-cetmix-bump-patch](https://github.com/cetmix/cetmix-tower/commits/17.0-ocabot-merge-pr-428-by-ivs-cetmix-bump-patch), awaiting test results.

**Comment by CetmixGitDrone** [comment]
Congratulations, your PR was merged at 3b38618a86b0bfc1954d89711422c9325b4fac83. Thanks a lot for contributing to cetmix. ❤️
